pkg/statedb/reflector/common.go

```diff
@@ -0,0 +1,13 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package reflector
+
+import (
+	"github.com/cilium/cilium/pkg/hive/cell"
+)
+
+// Reflector reflects external data into a statedb table
+type Reflector[Obj any] interface {
```

Comment:
Does this interface in combination with hive mean that we cannot reflect the same type from multiple sources?
I think it's probably fine, k8s types and kvstore types differ anyway, do you foresee an issue with that?

Reply:
This is really meant to be private to a module and currently it's only reason for existence is that we might need to be able to start and stop the reflector dynamically, so I don't see an issue with the types. And if there is, this can be wrapped.